File: cpp/problems/66.加一.cpp
// @before-stub-for-debug-begin
#include <vector>
#include <string>
#include "commoncppproblem66.h"

using namespace std;
// @before-stub-for-debug-end

/*
 * @lc app=leetcode.cn id=66 lang=cpp
 *
 * [66] 加一
 */

#include <iostream>
#include <vector>
using namespace std;

// @lc code=start
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        auto length = digits.size();
        for (auto i = length-1; i >= 0; i--) {
            if (digits[i] < 9) {
                digits[i]++; 
                return digits;
            }
            digits[i] = 0;
            if (i == 0) break;
        }
        digits.insert(digits.begin(), 1);
        return digits;
    }
};
// @lc code=end

File: go/src/leetcode/problems/66.加一.go
/*
 * @lc app=leetcode.cn id=66 lang=golang
 *
 * [66] 加一
 */

package problems

// @lc code=start
func plusOne(digits []int) []int {
	var length = len(digits)
	if length < 1 {
		return digits;
	}
	var i = length - 1
	for i >= 0 {
		if digits[i] < 9 {
			digits[i]++
			return digits
		}
		digits[i] = 0
		i--
	}
	var result []int
	result = append(result, 1)
	result = append(result, digits...)
	return result
}
// @lc code=end

